### PR TITLE
Kippt importer

### PR DESCRIPTION
**Pull request by andrewconner**
1) User uploads file directly to server (using ajax, to be implemented)
2) We parse the file with jsoup, grabbing the title, href, and tags
3) Create all tags that don't exist
4) Persist into RawKeep

Changes:
RawKeep has tagIds, which is a string like `3,100,1212` (comma separated Id[Collection]). RawKeepImporterPlugin uses these to auto-tag new keeps.

### CODE REVIEW

**Comment by andrewconner** [comment]
@stkem @leogrim Either of you mind reviewing the backend?

**Comment by andrewconner** [comment]
Grab me before merging, there's an evolution.

File: kifi-backend/modules/common/app/com/keepit/model/Keep.scala [hunk 1]
```diff
@@ -153,6 +153,7 @@ case class KeepSource(value: String) {
 object KeepSource {
   val keeper = KeepSource("keeper")
   val bookmarkImport = KeepSource("bookmarkImport")
+  val bookmarkFileImport = KeepSource("bookmarkFileImport")
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/model/Keep.scala:156]: Can you think of a way to identify the origin of the file (Kippt vs Delicious vs...)?

File: kifi-backend/modules/common/conf/evolutions/shoebox/165.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# SHOEBOX
+
+# --- !Ups
+
+alter TABLE raw_keep
+  add column tag_ids varchar(64) NULL;
```
Comment (leogrim) [inline kifi-backend/modules/common/conf/evolutions/shoebox/165.sql]: Is this long enough? A keep could have many tags if coming not from an actual bookmark but from another curation site like Delicious. Nope?

Comment (andrewconner) [inline kifi-backend/modules/common/conf/evolutions/shoebox/165.sql]: I can bump it, but it's IDs. You're right though, someone could easily have 20 tags, each with 6 digit IDs.

File: kifi-backend/modules/common/app/com/keepit/model/Keep.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/Keep.scala:156]: Only heuristically, it's iffy. The format is the same for all.

File: kifi-backend/modules/common/conf/evolutions/shoebox/165.sql (hunk 2 above)
Comment (stkem) [inline kifi-backend/modules/common/conf/evolutions/shoebox/165.sql]: you can just make it a `TEXT`

Comment (andrewconner) [inline kifi-backend/modules/common/conf/evolutions/shoebox/165.sql]: Done

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/RawKeepImporterPlugin.scala [hunk 3]
```diff
@@ -100,11 +101,36 @@ private class RawKeepImporterActor @Inject() (
...
           val tag = bookmarksCommanderProvider.get.getOrCreateTag(userId, tagName)(context)
-          bookmarksCommanderProvider.get.addToCollection(tag, successes)(context)
+          bookmarksCommanderProvider.get.addToCollection(tag.id.get, successes)(context)
         }
+
         //the bookmarks list may be very large!
         searchClient.updateURIGraph()
 
+        // Reduce all successes to a map of tagIdString -> tagId (typed), ignore errors (we c
+        val tagStrToId = successesRawKeep.map { rk =>
```
Comment (stkem) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/RawKeepImporterPlugin.scala]: I suggest giving `tagStrToId` an explicit type for clarity.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/RawKeepImporterPlugin.scala]: Done

File: kifi-backend/modules/common/app/com/keepit/model/Keep.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/Keep.scala:156]: Done, works for Kippt imports

**Comment by andrewconner** [comment]
Thanks @stkem and @leogrim

**Pull request merged**